gcloud-java-dns/src/test/java/com/google/cloud/dns/it/ITDnsTest.java

```diff
@@ -967,25 +967,920 @@ public void testListDnsRecords() {
+      assertNull(created.generatedId());
+      created = idResult.get();
+      assertEquals(ZONE1.name(), created.name()); // always returned
+      assertNull(created.creationTimeMillis());
+      assertNull(created.dnsName());
+      assertNull(created.description());
+      assertNotNull(created.nameServers());
+      assertTrue(created.nameServers().isEmpty()); // never returns null
+      assertNotNull(created.generatedId());
+      // combination of multiple things
+      created = combinationResult.get();
+      assertEquals(ZONE1.name(), created.name()); // always returned
+      assertNull(created.creationTimeMillis());
+      assertNull(created.dnsName());
+      assertEquals(ZONE1.description(), created.description());
+      assertFalse(created.nameServers().isEmpty());
+      assertNull(created.nameServerSet()); // we did not set it
+      assertNotNull(created.generatedId());
+    } finally {
+      DNS.delete(ZONE1.name());
+    }
+  }
+
+  @Test
+  public void testDeleteZoneBatch() {
+    try {
+      Zone created = DNS.create(ZONE1);
+      assertEquals(created, DNS.getZone(ZONE1.name()));
+      DnsBatch batch = DNS.batch();
+      DnsBatchResult<Boolean> result = batch.deleteZone(ZONE1.name());
+      batch.submit();
+      assertNull(DNS.getZone(ZONE1.name()));
+      assertTrue(result.get());
+    } finally {
+      DNS.delete(ZONE1.name());
+    }
+  }
+
+  @Test
+  public void testGetProjectBatch() {
+    // fetches all fields
+    DnsBatch batch = DNS.batch();
+    DnsBatchResult<ProjectInfo> result = batch.getProject();
+    DnsBatchResult<ProjectInfo> resultQuota =
+        batch.getProject(Dns.ProjectOption.fields(Dns.ProjectField.QUOTA));
+    DnsBatchResult<ProjectInfo> resultId =
+        batch.getProject(Dns.ProjectOption.fields(Dns.ProjectField.PROJECT_ID));
+    DnsBatchResult<ProjectInfo> resultNumber =
+        batch.getProject(Dns.ProjectOption.fields(Dns.ProjectField.PROJECT_NUMBER));
+    DnsBatchResult<ProjectInfo> resultCombination =
+        batch.getProject(Dns.ProjectOption.fields(Dns.ProjectField.PROJECT_NUMBER,
+            Dns.ProjectField.QUOTA, Dns.ProjectField.PROJECT_ID));
+    batch.submit();
+    assertNotNull(result.get().quota());
+    assertNotNull(resultQuota.get().quota());
+    assertNull(resultId.get().quota());
+    assertNull(resultNumber.get().quota());
+    assertNotNull(resultCombination.get().quota());
+  }
+
+  @Test
+  public void testCreateChangeBatch() {
+    try {
+      DNS.create(ZONE1, Dns.ZoneOption.fields(Dns.ZoneField.NAME));
+      DnsBatch batch = DNS.batch();
+      DnsBatchResult<ChangeRequest> result =
+          batch.applyChangeRequest(ZONE1.name(), CHANGE_ADD_ZONE1);
+      batch.submit();
+      ChangeRequest created = result.get();
+      assertEquals(CHANGE_ADD_ZONE1.additions(), created.additions());
+      assertNotNull(created.startTimeMillis());
+      assertTrue(created.deletions().isEmpty());
+      assertEquals("1", created.generatedId());
```

Comment:
Hm, I feel that we should only assertNotNull instead of checking for equality for generatedIds for two reasons:
1. The service owners could choose to change it (I don't know how likely that is)
2. We don't want to mislead users into thinking generated IDs will always follow this pattern.